File: snippets/dart_snippets/lib/lnurl_withdraw.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:breez_sdk/bridge_generated.dart';

Future<void> lnurlWithdraw() async {
  // ANCHOR: lnurl-withdraw
  /// Endpoint can also be of the form:
  /// lnurlw://domain.com/lnurl-withdraw?key=val
  String lnurlWithdrawUrl =
      "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4exctthd96xserjv9mn7um9wdekjmmw843xxwpexdnxzen9vgunsvfexq6rvdecx93rgdmyxcuxverrvcursenpxvukzv3c8qunsdecx33nzwpnvg6ryc3hv93nzvecxgcxgwp3h33lxk";

  InputType inputType = await breezSDK.parseInput(input: lnurlWithdrawUrl);
  if (inputType is InputType_LnUrlWithdraw) {
    int amountMsat = inputType.data.minWithdrawable;
    LnUrlWithdrawRequest req = LnUrlWithdrawRequest(
      data: inputType.data,
      amountMsat: amountMsat,
      description: "<description>",
    );
    LnUrlWithdrawResult result = await breezSDK.lnurlWithdraw(req: req);
    print(result.data);
  }
  // ANCHOR_END: lnurl-withdraw
}

File: snippets/react-native/lnurl_withdraw.ts
import {
    InputTypeVariant,
    parseInput,
    withdrawLnurl
} from "@breeztech/react-native-breez-sdk"

const exampleLnurlWithdraw = async () => {
    // ANCHOR: lnurl-withdraw
    // Endpoint can also be of the form:
    // lnurlw://domain.com/lnurl-withdraw?key=val
    const lnurlWithdrawUrl =
        "lnurl1dp68gurn8ghj7mr0vdskc6r0wd6z7mrww4exctthd96xserjv9mn7um9wdekjmmw843xxwpexdnxzen9vgunsvfexq6rvdecx93rgdmyxcuxverrvcursenpxvukzv3c8qunsdecx33nzwpnvg6ryc3hv93nzvecxgcxgwp3h33lxk"

    const input = await parseInput(lnurlWithdrawUrl)
    if (input.type === InputTypeVariant.LN_URL_WITHDRAW) {
        const amountMsat = input.data.minWithdrawable
        const lnUrlWithdrawResult = await withdrawLnurl({
            data: input.data,
            amountMsat,
            description: "comment"
        })
    }
    // ANCHOR_END: lnurl-withdraw
}
